x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/TransportGetProfilingAction.java

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.profiler;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.get.MultiGetItemResponse;
+import org.elasticsearch.action.get.MultiGetResponse;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.util.Maps;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.search.aggregations.bucket.terms.StringTerms;
+import org.elasticsearch.search.aggregations.bucket.terms.TermsAggregationBuilder;
+import org.elasticsearch.search.aggregations.metrics.Sum;
+import org.elasticsearch.search.aggregations.metrics.SumAggregationBuilder;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.profiler.utils.MapExtractor;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+public class TransportGetProfilingAction extends HandledTransportAction<GetProfilingRequest, GetProfilingResponse> {
+    private final NodeClient nodeClient;
+    private final ThreadContext threadContext;
+
+    @Inject
+    public TransportGetProfilingAction(TransportService transportService, ActionFilters actionFilters, NodeClient nodeClient) {
+        super(GetProfilingAction.NAME, transportService, actionFilters, GetProfilingRequest::new);
+        this.nodeClient = nodeClient;
+        this.threadContext = transportService.getThreadPool().getThreadContext();
+    }
+
+    @Override
+    protected void doExecute(Task submitTask, GetProfilingRequest request, ActionListener<GetProfilingResponse> submitListener) {
+        try (var ignored = threadContext.newTraceContext()) {
+            EventsIndex mediumDownsampled = EventsIndex.MEDIUM_DOWNSAMPLED;
+            this.nodeClient.prepareSearch(mediumDownsampled.getName())
```

Comment:
Is this index always local or could it be a remote index? In general, I am assuming that this API will not support cross-cluster execution, but maybe it's a good question to ask.

Reply:
Indeed, this index is always local at the moment and there are at least no immediately plans to support cross-cluster search. https://github.com/elastic/elasticsearch/issues/90129 would be required to issue the subsequent `mget` calls in a remote cluster. 

If I understand the reference docs correctly, we'd need to combine the names of the remote clusters and the index name(s) into an index pattern in order to run this search request against multiple clusters?

Reply:
yes a search with index `remote:index` goes to the remote cluster configured with `remote` alias.

Reply:
Thanks for the confirmation!